src/utils/reactUtils.tsx

```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
+
+// These functions are migrated from the now deprecated recompose package.
+// see https://github.com/acdlite/recompose/blob/master/src/packages/recompose/withProps.js
+type mapper<TInner, TOutter> = (input: TInner) => TOutter;
+
+// Injects props and removes them from the prop requirements.
+// Will not pass through the injected props if they are passed in during
+// render. Also adds new prop requirements from TNeedsProps.
+type InferableComponentEnhancerWithProps<TInjectedProps, TNeedsProps> = <
+	P extends TInjectedProps,
+>(
+	component: React.ComponentType<P>,
+) => React.ComponentClass<Omit<P, keyof TInjectedProps> & TNeedsProps>;
+
+// Accepts a function that merges owner props with a new collection of props
+// that are then passed to the base component.
+export function withProps<TInner, TOutter>(
+	fn: TInner | mapper<TOutter, TInner>,
+): InferableComponentEnhancerWithProps<TInner & TOutter, TOutter> {
+	const wrapper = (BaseComponent: any) => {
+		const hoc = (props = {}) => {
+			const combinedProps = {
+				...props,
+				...(typeof fn === 'function' ? (fn as any)(props) : fn),
+			};
+
+			return <BaseComponent {...combinedProps} />;
```

Comment:
That removed an unnecessary destructuring I think :thinking: 
```suggestion
			return <BaseComponent
				{...props}
				{...(typeof fn === 'function' ? (fn as any)(props) : fn)}
			/>;
```

Reply:
Good call, I'll change this